### PR TITLE
[CAMEL-15894] - Update to opentelemetry 0.11

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

This PR is about updating opentelemetry to the most recent version (at this time is 0.11)

- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: components/camel-opentelemetry/pom.xml [hunk 1]
```diff
@@ -80,7 +75,7 @@
     </dependency>
     <dependency>
       <groupId>io.opentelemetry</groupId>
-      <artifactId>opentelemetry-exporters-inmemory</artifactId>
+      <artifactId>opentelemetry-sdk-testing</artifactId>
```
Comment (oscerd) [inline components/camel-opentelemetry/pom.xml:78]: Is this a test scope dependency?

**Review by oscerd — CHANGES_REQUESTED** [review]
Just a question. Please add documentation to migration guide and into adoc. Also can you double check the camel-karaf feature in camel-karaf repo?

File: components/camel-opentelemetry/pom.xml (hunk 1 above)
Comment (rubenvp8510) [reply components/camel-opentelemetry/pom.xml:78]: Yes.

Comment (oscerd) [reply components/camel-opentelemetry/pom.xml:78]: It makes sense to add scope test then

**Comment by rubenvp8510** [comment]
Wondering why other modules have style failures, I haven't touched it.

**Comment by rubenvp8510** [comment]
@oscerd I'm not familiar with  camel-karaf feature. Could you provide me some guiadance on what I need to check? 
Thanks!

**Comment by oscerd** [comment]
@rubenvp8510 the feature related to this component lives here: https://github.com/apache/camel-karaf/blob/master/platforms/karaf/features/src/main/resources/features.xml#L1997

To check you need to fully build the main camel repository and on the camel-karaf repo run mvn clean install in the platforms/karaf/features folder. There will be a verification of the features and you'll eventually get errors. 

By the way we can also check it later if you don't have time.

**Comment by rubenvp8510** [comment]
@oscerd I reviewed and sent a PR to karaf repository: https://github.com/apache/camel-karaf/pull/53

**Review by davsclaus — APPROVED** [review]

**Pull request merged**